go/p2p/host.go

```diff
@@ -102,6 +102,7 @@ func (cfg *HostConfig) Load() error {
 
 // ConnManagerConfig describes a set of settings for a connection manager.
 type ConnManagerConfig struct {
+	// XXX: MinPeers in main config is uint32, but int here -- investigate.
```

Comment:
Does this need investigation?

Reply:
Yes (same for other type mismatches in the new p2p stack).

I imagine that the underlying library we're using has the signed types -- in this case, we should also change the config to use signed types and check that they don't overflow, since it's safer than blindly casting an uint to an int.

Reply:
I've modified this as described above.